src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -54,6 +55,13 @@ internal partial class RequestExecutionQueue
         private readonly CancellationTokenSource _cancelSource;
         private readonly DocumentChangeTracker _documentChangeTracker;
 
+        // These two dictionaries are used to cache our forked LSP solution so we don't have to
+        // recompute it for each request. We don't need to worry about threading because they are only
+        // used when preparing to handle a request, which happens in a single thread in the ProcessQueueAsync
+        // method.
+        private readonly Dictionary<Workspace, Solution> _lastWorkspaceSolution = new();
+        private readonly Dictionary<Workspace, Solution> _lastLSPSolution = new();
```

Comment:
As per https://github.com/dotnet/roslyn/pull/49305/files#r522428751, i think this will be cleaner and simpler if this is one dictionary of `Dictionary<Workspace, (Solution lastWorkspaceSolution, Solution lastLspSolution)>`